iree/runtime/demo/BUILD

```diff
@@ -19,7 +19,8 @@ package(
 
 iree_cmake_extra_content(
     content = """
-if (NOT ${IREE_BUILD_COMPILER} OR NOT ${IREE_BUILD_TESTS})
+if (NOT ${IREE_HAL_DRIVER_VMVX} OR NOT ${IREE_BUILD_COMPILER}
+    OR NOT ${IREE_BUILD_TESTS})
```

Comment:
Technically this should check for `IREE_HAL_DRIVER_VMVX` and `IREE_TARGET_BACKEND_VMVX` (or more specifically the existence of `iree_runtime_testdata_simple_mul_module_c`), then `iree/runtime/testdata/` should check for `IREE_TARGET_BACKEND_VMVX`

checking for `IREE_BUILD_COMPILER` is not necessary, since any `IREE_TARGET_BACKEND_*` implies `IREE_BUILD_COMPILER`:
https://github.com/google/iree/blob/7bc297e9fc8bcc7f291d84e8854a44d5161f2cd6/CMakeLists.txt#L151-L159

Reply:
Good point. Condition updated.